### DISCUSSION TITLE
v1.4.4 程序无法使用

### DISCUSSION THREAD

**Discussion by Korlne** [KIvoy/nikke-assistant#12] [Q&A] [answered]
助理1.4.4版本, 可以识别游戏窗口，测试多个功能（领取前哨奖励，咨询妮姬，新秀竞技场）均无法使用。

其中选择新秀竞技场后返回游戏界面会提示rookie arena ark icon is not found, retrying at home，无限重复返回操作，Del无法结束任务，同时回到助理窗口后助理会未响应。

游戏为简体中文界面，NIKKE_ASSISTANT.INI 内设置均为 zh。

无app_err 文件生成，只有app_log
[app_log.txt](https://github.com/KIvoy/nikke-assistant/files/14642629/app_log.txt)

**Comment by KIvoy** [comment]
你好， 我看了一下log， 发现问题在于基本上无法识别每个功能的对应图像。几个问题能帮助解答：
1. 你之前用过1.4.4版本么？能用么？
2. 如果之前能用现在不能用， 你还记得什么时候开始的？
3. 如果没有用过， 请确认助手确实识别了正确的窗口， 因为有时候会误识别同名的窗口。 你可以用“调整分辨率”的功能来测试一下， 如果确实改变了游戏的窗口大小就说明识别成功了。大部分类似问题都是因为窗口识别错误或者窗口分辨率不正确而导致的。记得在确认识别正确后用”重新加载游戏“来还原成默认分辨率（591x1061)。

**Reply by Korlne** [reply]
你好，感谢回答

我这是我第一次使用NIKKE_ASSISTANT

- 在打开pc版Nikke后开启助手，原本的窗口会被助手自动调整到591x1061，并且“选择分辨率“可以改变窗口大小，所以应该是识别到了正确的窗口

- 刚经过测试尽管通过“选择分辨率”可以改变窗口大小，但使用"领取前哨奖励"功能仍然无法识别到对应按键

![Uploading Nikke.png…]()


- 并且发现当同时在桌面打开Nikke游戏窗口和Nikke启动器时，在助手“选择游戏窗口”里可以看到两个Nikke进程，此时选择任何两个Nikke进程改变分辨率都无法正确识别到Nikke游戏窗口，而是取决于上一次活跃的是游戏窗口还是启动器窗口（例如上一次鼠标在启动器窗口点击过后直接移动到助手界面，此时不管选哪个Nikke进程来进行“选择分辨率”都会改变启动器窗口的窗口大小）

- 将启动器窗口关闭后只会显示一个Nikke进程，此时“选择分辨率”可以改变窗口大小，却仍然无法识别功能对应的图像

- 系统版本为Win11 专业版 23H2

**Reply by KIvoy** [reply]
谢谢你详细的描述， 根据描述来看， 你可以尝试一下以下的解决方法：
1. 试着把所有画质调到最高。 因为用于识别的截图全部使用的是最高画质， 所以画质的问题有可能导致识别失败。
2. 虽然不太可能， 但是请保证使用的显示器至少是1920x1080， 并且游戏窗口没有被遮挡。

如果以上方法都不行， 那Win11可能是最大问题。 由于程序是在win10系统开发测试的， 并未在win11测试。 由于此前也有朋友提交关于win11的报错， 系统版本可能是造成问题的关键。 

可惜的是， 我当前没有时间以及环境来测试和匹配win11, 所以很抱歉可能短时间内解决不了这个问题。你可以尝试着用win10的系统来测试下是否是这个问题。

**Reply by Korlne** [reply]
谢谢，画质已确认是最高，显示器2560x1440，并窗口无遮挡。

基本上可以认为是win11问题了

**Reply by Korlne** [reply]
悲报，委托了一位同样是Win11  专业版23H2的朋友进行测试，结果他可以正常使用 :(

暂无发现其他导致问题的原因

**Comment by Korlne** [comment, marked as answer]
已发现问题解决方法

- 问题原因：使用助手时有两个屏幕，导致助手无法正确选取位于主屏的游戏窗口

- 解决过程：关闭副屏，随后助手可以正确识别

- 残余问题：在与朋友测试时发现，他也同时有两个屏幕却可以正常使用，而我这边需要关闭副屏才能正常使用
                     朋友的主屏分辨率为1920*1080，我的主屏为2560x1440 副屏为1920*1080，与两屏幕不一致，将主副屏分辨率调整一致后仍 
                     无法识别


- 可能的建议：增加识别游戏窗口位于的屏幕后再进行正常的图像识别进程

**Reply by KIvoy** [reply]
原来如此， 之前也有人遇到过多显示屏的问题， 不过我以为已经完全修复了。 从理论上来说这里的抓图是全部显示屏一起抓的。 由于我无法复现该问题， 所以请问是不管游戏窗口在主屏还是副屏都无法识别么？ 

最后， 很感谢你的辛苦测试， 等我有时间了再看看如何解决。 当然如果你能在你的环境内发现问题并提交PR, 我也会感激不尽。

**Reply by Korlne** [reply]
是的，特意测试过把游戏窗口分别放到主屏和副屏，均无法选取窗口。但就像我描述的，我朋友在和我有相似的环境下却可以正常识别。

如果我能定位到问题的话会提交PR的。

感谢你一直跟进这个问题 :)
